--- a/libsolidity/codegen/ir/Common.h
+++ b/libsolidity/codegen/ir/Common.h
@@ -42,6 +42,9 @@ struct YulArity
 	bool operator==(YulArity const& _other) const { return in == _other.in && out == _other.out; }
 	bool operator!=(YulArity const& _other) const { return !(*this == _other); }
+	bool operator<(YulArity const& _other) const { return in < _other.in || (in == _other.in && out < _other.out); }
+	bool operator>(YulArity const& _other) const { return _other < *this; }
+	bool operator<=(YulArity const& _other) const { return !(_other < *this); }
 
 	size_t in;  /// Number of input parameters
 	size_t out; /// Number of output parameters
