diff --git a/mk-ca-bundle.pl b/mk-ca-bundle.pl
index 926478f..51db464 100644
--- a/mk-ca-bundle.pl
+++ b/mk-ca-bundle.pl
@@ -407,8 +407,6 @@ print CRT <<EOT;
 ##
 ## Bundle of CA Root Certificates
 ##
-## Certificate data from Mozilla ${datesrc}: ${currentdate} GMT
-##
 ## This is a bundle of X.509 certificates of public Certificate Authorities
 ## (CA). These were automatically extracted from Mozilla's root certificates
 ## file (certdata.txt).  This file can be found in the mozilla source tree:
