diff --git a/src/common.hpp b/src/common.hpp
index 7b662b87..1a413e0c 100644
--- a/src/common.hpp
+++ b/src/common.hpp
@@ -195,7 +195,7 @@ template <typename T>
 struct LList
 {
     const LList*  m_prev;
-    T   m_item;
+    T   m_item{};
 
     LList():
         m_prev(nullptr)
diff --git a/src/hir_typeck/static.cpp b/src/hir_typeck/static.cpp
index 2c32adf4..c14f4bc2 100644
--- a/src/hir_typeck/static.cpp
+++ b/src/hir_typeck/static.cpp
@@ -3608,7 +3608,7 @@ StaticTraitResolve::ValuePtr StaticTraitResolve::get_value(const Span& sp, const
                 }
 
                 const auto& impl_params = out_params.pp_impl_data;
-                for(size_t i = 0; i < impl_params.m_types.size(); i ++ ) {
+                /*for(size_t i = 0; i < impl_params.m_types.size(); i ++ ) {
                     if( !params_set.m_types[i] ) {
                         // TODO: Error when there's a type param that can't be determined?
                     }
@@ -3622,7 +3622,7 @@ StaticTraitResolve::ValuePtr StaticTraitResolve::get_value(const Span& sp, const
                     if( !params_set.m_lifetimes[i] ) {
                         // TODO: Error when there's a lifetime param that can't be determined?
                     }
-                }
+                }*/
 
                 out_params.pp_impl = &out_params.pp_impl_data;
                 DEBUG("PP impl = " << *out_params.pp_impl);
