diff --git a/Makefile b/Makefile
index 4317d2c0..efa9bf1b 100644
--- a/Makefile
+++ b/Makefile
@@ -170,11 +170,11 @@ $(BIN): $(OBJDIR)main.o bin/mrustc.a bin/common_lib.a
 	@+mkdir -p $(dir $@)
 	@echo [CXX] -o $@
 ifeq ($(OS),Windows_NT)
-	$V$(CXX) -o $@ $(LINKFLAGS) $(OBJDIR)main.o -Wl,--whole-archive bin/mrustc.a bin/common_lib.a -Wl,--no-whole-archive $(LIBS)
+	$V$(CXX) -o $@ $(LINKFLAGS) $(OBJDIR)main.o -Wl,--whole-archive bin/mrustc.a bin/common_lib.a -Wl,--no-whole-archive $(LIBS) $(CXXFLAGS_EXTRA)
 else ifeq ($(shell uname -s || echo not),Darwin)
-	$V$(CXX) -o $@ $(LINKFLAGS) $(OBJDIR)main.o -Wl,-all_load bin/mrustc.a bin/common_lib.a $(LIBS)
+	$V$(CXX) -o $@ $(LINKFLAGS) $(OBJDIR)main.o -Wl,-all_load bin/mrustc.a bin/common_lib.a $(LIBS) $(CXXFLAGS_EXTRA)
 else
-	$V$(CXX) -o $@ $(LINKFLAGS) $(OBJDIR)main.o -Wl,--whole-archive bin/mrustc.a -Wl,--no-whole-archive bin/common_lib.a $(LIBS)
+	$V$(CXX) -o $@ $(LINKFLAGS) $(OBJDIR)main.o -Wl,--whole-archive bin/mrustc.a -Wl,--no-whole-archive bin/common_lib.a $(LIBS) $(CXXFLAGS_EXTRA)
 	$(OBJCOPY) --only-keep-debug $(BIN) $(BIN).debug
 	$(OBJCOPY) --add-gnu-debuglink=$(BIN).debug $(BIN)
 	$(STRIP) $(BIN)
diff --git a/src/trans/codegen_c.cpp b/src/trans/codegen_c.cpp
index 277fc054..afc83d2c 100644
--- a/src/trans/codegen_c.cpp
+++ b/src/trans/codegen_c.cpp
@@ -1294,17 +1294,15 @@ namespace {
                     args.push_back("-O1");  // HACK: Work around mrustc #347 by reducing the optimisation level
                     break;
                 }
-                // HACK: Work around [https://gcc.gnu.org/bugzilla/show_bug.cgi?id=117423] by disabling an optimisation stage
-                if( opt.opt_level > 0 )
-                {
-                    args.push_back("-fno-tree-sra");
-                }
                 if( opt.emit_debug_info )
                 {
                     args.push_back("-g");
                 }
                 // TODO: Why?
                 args.push_back("-fPIC");
+                args.push_back("-L/usr/lib/llvm17/lib");
+                args.push_back("-I/usr/lib/llvm17/include");
+                args.push_back("-fuse-ld=/usr/lib/llvm17/bin/ld.lld");
                 args.push_back("-o");
                 switch(out_ty)
                 {
@@ -5442,7 +5440,7 @@ namespace {
                         m_of << "+";
                     }
                     else {
-                        m_of << (p.input ? "+" : "=");
+                        m_of << (p.input ? "=" : "=");
                     }
                     TU_MATCH_HDRA((p.spec), {)
                     TU_ARMA(Class, c)
diff --git a/tools/minicargo/Makefile b/tools/minicargo/Makefile
index 4c713d2d..aed6c187 100644
--- a/tools/minicargo/Makefile
+++ b/tools/minicargo/Makefile
@@ -40,7 +40,7 @@ clean:
 $(BIN): $(OBJS) ../../bin/common_lib.a
 	@+mkdir -p $(dir $@)
 	@echo [CXX] -o $@
-	$V$(CXX) -o $@ $(OBJS) ../../bin/common_lib.a $(LINKFLAGS)
+	$V$(CXX) -o $@ $(OBJS) ../../bin/common_lib.a $(LINKFLAGS) $(CXXFLAGS_EXTRA)
 ifeq ($(OS),Windows_NT)
 else ifeq ($(shell uname -s || echo not),Darwin)
 else
