diff --git a/src/elfcompress.c b/src/elfcompress.c
--- a/src/elfcompress.c
+++ b/src/elfcompress.c
@@ -36,6 +36,10 @@
 #include <gelf.h>
 #include "system.h"
 #include "libeu.h"
+
+#ifndef FNM_EXTMATCH
+# define FNM_EXTMATCH 0
+#endif
 #include "printversion.h"
 
 /* Really should come from libgen.h, but we poisoned basename in system.h.  */

diff --git a/src/strip.c b/src/strip.c
--- a/src/strip.c
+++ b/src/strip.c
@@ -30,6 +30,10 @@
 #include <libelf.h>
 #include <locale.h>
 #include <stdbool.h>
+
+#ifndef FNM_EXTMATCH
+# define FNM_EXTMATCH 0
+#endif
 #include <stdio.h>
 #include <stdio_ext.h>
 #include <stdlib.h>
