diff --git a/helpers.sh b/helpers.sh
index ea2cc05..3429e61 100755
--- a/helpers.sh
+++ b/helpers.sh
@@ -32,7 +32,7 @@ _get_files() {
         # Archive symlinks to directories as symlinks
         echo "${prefix}/${f}"
         if [ -d "${f}" ] && ! [ -h "${f}" ]; then
-            cd "${f}"
+            cd -- "${f}"
             _get_files "${prefix}/${f}"
             cd ..
         fi
@@ -56,7 +56,7 @@ reset_timestamp() {
         for f in ${fs}; do
             touch -h -t 197001010000.00 "${f}"
             if [ -d "${f}" ]; then
-                cd "${f}"
+                cd -- "${f}"
                 reset_timestamp
                 cd ..
             fi
