diff --git a/awk.h b/awk.h
index dbad0d810..668a7bc41 100644
--- a/awk.h
+++ b/awk.h
@@ -406,17 +406,20 @@ typedef struct exp_node {
 		} nodep;
 
 		struct {
-#ifdef HAVE_MPFR
 			union {
 				AWKNUM fltnum;
+#ifdef HAVE_MPFR
 				mpfr_t mpnum;
 				mpz_t mpi;
+#else
+#if SIZEOF_VOID_P == 4
+				char alignment[28];
+#else	// SIZEOF_VOID_P != 4
+				char alignment[48];
+#endif	// SIZEOF_VOID_P != 4
+#endif	// HAVE_MPFR
 			} nm;
 			int rndmode;
-#else
-			AWKNUM fltnum;
-			int for_alignment_only;	// especially on 32-bit
-#endif
 			char *sp;
 			size_t slen;
 			int idx;
@@ -561,10 +564,8 @@ typedef struct exp_node {
 #ifdef HAVE_MPFR
 #define mpg_numbr	sub.val.nm.mpnum
 #define mpg_i		sub.val.nm.mpi
-#define numbr		sub.val.nm.fltnum
-#else
-#define numbr		sub.val.fltnum
 #endif
+#define numbr		sub.val.nm.fltnum
 #define typed_re	sub.val.typre
 
 /*
