Index: bin/compile
<+>UTF-8
===================================================================
diff --git a/bin/compile b/bin/compile
--- a/bin/compile	(revision 29e47ed68cea82a0006c1c8949d85c7b15b11a0e)
+++ b/bin/compile	(date 1746221810904)
@@ -4,7 +4,7 @@
 $cwd = getcwd();
 assert(is_string($cwd));
 chdir(__DIR__.'/../');
-$ts = rtrim(exec('git rev-list --no-commit-header -n1 --format=%ct HEAD'));
+$ts = intval(getenv('SOURCE_DATE_EPOCH'));
 if (!is_numeric($ts)) {
     echo 'Could not detect date using "git rev-list --no-commit-header -n1 --format=%ct HEAD"'.PHP_EOL;
     exit(1);
