Index: src/Composer/Compiler.php
<+>UTF-8
===================================================================
diff --git a/src/Composer/Compiler.php b/src/Composer/Compiler.php
--- a/src/Composer/Compiler.php
+++ b/src/Composer/Compiler.php
@@ -49,37 +49,9 @@ class Compiler
             unlink($pharFile);
         }

-        $process = new ProcessExecutor();
-
-        $command = Git::buildRevListCommand($process, ['-n1', '--format=%H', 'HEAD']);
-        if (0 !== $process->execute($command, $output, dirname(__DIR__, 2))) {
-            throw new \RuntimeException('Can\'t run git rev-list. You must ensure to run compile from composer git repository clone and that git binary is available.');
-        }
-        $this->version = trim(Git::parseRevListOutput($output, $process));
-
-        $command = Git::buildRevListCommand($process, ['-n1', '--format=%ci', 'HEAD']);
-        if (0 !== $process->execute($command, $output, dirname(__DIR__, 2))) {
-            throw new \RuntimeException('Can\'t run git rev-list. You must ensure to run compile from composer git repository clone and that git binary is available.');
-        }
-
-        $this->versionDate = new \DateTime(trim(Git::parseRevListOutput($output, $process)));
-        $this->versionDate->setTimezone(new \DateTimeZone('UTC'));
-
-        if (0 === $process->execute(['git', 'describe', '--tags', '--exact-match', 'HEAD'], $output, dirname(__DIR__, 2))) {
-            $this->version = trim($output);
-        } else {
-            // get branch-alias defined in composer.json for dev-main (if any)
-            $localConfig = __DIR__.'/../../composer.json';
-            $file = new JsonFile($localConfig);
-            $localConfig = $file->read();
-            if (isset($localConfig['extra']['branch-alias']['dev-main'])) {
-                $this->branchAliasVersion = $localConfig['extra']['branch-alias']['dev-main'];
-            }
-        }
-
-        if ('' === $this->version) {
-            throw new \UnexpectedValueException('Version detection failed');
-        }
+        $this->versionDate = new \DateTime();
+        $this->versionDate->setTimestamp(intval(getenv('SOURCE_DATE_EPOCH')));
+        $this->version = getenv('STAGEX_COMPOSER_VERSION');

         $phar = new \Phar($pharFile, 0, 'composer.phar');
         $phar->setSignatureAlgorithm(\Phar::SHA512);
