--- a/scripts/gen-randstruct-seed.sh
+++ b/scripts/gen-randstruct-seed.sh
@@ -1,7 +1,7 @@
 #!/bin/sh
 # SPDX-License-Identifier: GPL-2.0
 
-SEED=$(od -A n -t x8 -N 32 /dev/urandom | tr -d ' \n')
+if [ -f "$1" ]; then SEED=$(cat "$1"); else SEED=$(od -A n -t x8 -N 32 /dev/urandom | tr -d " \n"); fi
 echo "$SEED" > "$1"
 HASH=$(echo -n "$SEED" | sha256sum | cut -d" " -f1)
 echo "#define RANDSTRUCT_HASHED_SEED \"$HASH\"" > "$2"
