From: Stagex Build System
Date: 2026-03-30
Subject: Use system eigen instead of downloading

--- a/third-party/eigen/CMakeLists.txt
+++ b/third-party/eigen/CMakeLists.txt
@@ -3,17 +3,13 @@
 # Copyright Advanced Micro Devices, Inc.
 # SPDX-License-Identifier: MIT
 
-set(_source_dir "${CMAKE_CURRENT_BINARY_DIR}/source")
-set(_download_stamp "${_source_dir}/download.stamp")
-
-therock_subproject_fetch(eigen-sources
-  SOURCE_DIR "${_source_dir}"
-  # Originally mirrored from: "https://gitlab.com/libeigen/eigen/-/archive/3.4.0/eigen-3.4.0.tar.bz2"
-  URL "https://rocm-third-party-deps.s3.us-east-2.amazonaws.com/eigen-3.4.0.tar.bz2"
-  URL_HASH "SHA256=5630c0c6f43f26790a677a863f1641da4c014a8e5e9201a03494d970b12579e0"
-  TOUCH "${_download_stamp}"
-)
+# Use system eigen
+find_package(Eigen3 3.4 REQUIRED NO_MODULE)
 
-therock_cmake_subproject_declare(therock-eigen
-  EXTERNAL_SOURCE_DIR "eigen"
-  USE_DIST_AMDGPU_TARGETS
-  NO_MERGE_COMPILE_COMMANDS
-  BACKGROUND_BUILD
-  OUTPUT_ON_FAILURE
-  EXTRA_DEPENDS "${_download_stamp}"
-)
+# Provide package for downstream dependencies
+set(EIGEN_SOURCE_DIR "/usr/include/eigen3")
+
+therock_cmake_subproject_declare(therock-eigen
+  EXTERNAL_SOURCE_DIR "${EIGEN_SOURCE_DIR}"
+  USE_DIST_AMDGPU_TARGETS
+  NO_MERGE_COMPILE_COMMANDS
+)
