From: Stagex Build System
Date: 2026-03-30
Subject: Use system flatbuffers, fmt, googletest instead of downloading

--- a/third-party/flatbuffers/CMakeLists.txt
+++ b/third-party/flatbuffers/CMakeLists.txt
@@ -1,15 +1,12 @@
 # Copyright Advanced Micro Devices, Inc.
 # SPDX-License-Identifier: MIT
 
-therock_subproject_fetch(therock-flatbuffers-sources
-  CMAKE_PROJECT
-  # Originally mirrored from: https://github.com/google/flatbuffers/archive/refs/tags/v25.9.23.tar.gz
-  URL https://rocm-third-party-deps.s3.us-east-2.amazonaws.com/flatbuffers-25.9.23.tar.gz
-  URL_HASH SHA256=9102253214dea6ae10c2ac966ea1ed2155d22202390b532d1dea64935c518ada
-)
+# Use system flatbuffers
+find_package(flatbuffers REQUIRED)
 
 therock_cmake_subproject_declare(therock-flatbuffers
-  USE_DIST_AMDGPU_TARGETS
-  BACKGROUND_BUILD
-  EXCLUDE_FROM_ALL
-  NO_MERGE_COMPILE_COMMANDS
-  OUTPUT_ON_FAILURE
-  EXTERNAL_SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/source"
+  USE_DIST_AMDGPU_TARGETS
   CMAKE_ARGS
     -DFLATBUFFERS_BUILD_TESTS=OFF
     -DFLATBUFFERS_BUILD_FLATC=ON
 )
