diff --git a/mk-ca-bundle.pl b/mk-ca-bundle.pl
index 79c78eb..80de253 100644
--- a/mk-ca-bundle.pl
+++ b/mk-ca-bundle.pl
@@ -401,8 +401,6 @@ print CRT <<EOT;
 ##
 ## Bundle of CA Root Certificates
 ##
-## Certificate data from Mozilla ${datesrc}: ${currentdate} GMT
-##
 ## Find updated versions here: https://curl.se/docs/caextract.html
 ##
 ## This is a bundle of X.509 certificates of public Certificate Authorities
