diff --git a/setup.py b/setup.py
index 5d9fd285..5c86ad03 100644
--- a/setup.py
+++ b/setup.py
@@ -60,6 +60,6 @@ else:
 setup(
     version=version,
     cmdclass={"bdist_wheel": bdist_wheel},
-    rust_extensions=[RustBin("maturin", args=cargo_args, cargo_manifest_args=["--locked"], env=env)],
+    rust_extensions=[RustBin("maturin", args=cargo_args, cargo_manifest_args=["--frozen" if "--frozen" in cargo_args else "--locked"], env=env)],
     zip_safe=False,
 )
